rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -2281,11 +2281,23 @@ extern "C"
 		return RMW_RET_ERROR;
 	}
 	//Get participant pointer from node
-        if(node->implementation_identifier != eprosima_fastrtps_identifier)
-        {
-            RMW_SET_ERROR_MSG("node handle not from this implementation");
-            return RMW_RET_ERROR;
-        }
+	if(node->implementation_identifier != eprosima_fastrtps_identifier)
+	{
+	    RMW_SET_ERROR_MSG("node handle not from this implementation");
+	    return RMW_RET_ERROR;
+	}
+	if (topic_names_and_types->topic_count) {
```

Comment:
Yeah I replicated the behavior of the other rmw impl. we can always improve it in a follow-up PR if we settle on a better API for this